rom/dev/README.md

```diff
@@ -242,50 +254,85 @@ Both UDS and Field Entropy are available only during cold reset of Caliptra.
 
-2. Clear the UDS in key vault
+    If Dual-Cert Mode is enabled, use the conditioned UDS to derive the MLDSA CDI store the resultant mac in Key Vault Slot 12
+
+    `hmac512_kdf(KvSlot10, b"idevid_cdi", KvSlot12)`
+
+2. Clear the UDS(s) in key vault
 
     `kv_clear(KvSlot0)`
 
-3. Derive ECC Key Pair using CDI in Key Vault Slot6 and store the generated private key in KeySlot7
+    If Dual-Cert Mode is enabled:
+
+    `kv_clear(KvSlot10)`
+
+3. Derive ECC Key Pair using CDI in Key Vault Slot 6 and store the generated private key in Key Vault Slot 7
 
     `IDevIDSeed = hmac384_kdf(KvSlot6, b"idevid_keygen", KvSlot3)`
-    `IDevIdPubKey = ecc384_keygen(KvSlot3, KvSlot7)`
+
+    `IDevIdPubKeyEcdsa = ecc384_keygen(KvSlot3, KvSlot7)`
+
     `kv_clear(KvSlot3)`
 
-*(Note: Steps 4-7 are performed if CSR download is requested via CPTRA_DBG_MANUF_SERVICE_REG register)*
+    If Dual-Cert Mode is enabled, derive the MLDSA Key Pair using the conditioned CDI in Key Vault Slot 12.
+
+    `IDevIDSeed = hmac512_kdf(KvSlot12, b"idevid_keygen", KvSlot10)`
+
+    `IDevIdPubKeyMldsa = mldsa87_keygen(KvSlot10)`
 
-4. Generate the `To Be Signed` DER Blob of the IDevId CSR
+*(Note: Steps 4-11 are performed if CSR download is requested via CPTRA_DBG_MANUF_SERVICE_REG register)*
 
-    `IDevIdTbs = gen_tbs(IDEVID_CSR, IDevIdPubKey)`
+4. Generate the `To Be Signed` DER Blob of the IDevId CSR with the ECDSA public key.
 
-5. Sign the IDevID `To Be Signed` DER Blob with IDevId Private Key in Key Vault Slot 7
+    `IDevIdTbsEcdsa = gen_tbs(IDEVID_CSR, IDevIdPubKeyEcdsa)`
 
-    `IDevIdTbsDigest = sha384_digest(IDevIdTbs)`
-    `IDevIdCertSig = ecc384_sign(KvSlot7, IDevIdTbsDigest)`
+5. Sign the IDevID `To Be Signed` DER Blob with IDevId ECDSA Private Key in Key Vault Slot 7
+
+    `IDevIdTbsDigestEcdsa = sha384_digest(IDevIdTbsEcdsa)`
+
+    `IDevIdCertSigEcdsa = ecc384_sign(KvSlot7, IDevIdTbsDigestEcdsa)`
 
 6. Verify the signature of IDevID `To Be Signed` Blob
 
-    `IDevIdTbsDigest = sha384_digest(IDevIdTbs)`
-    `Result = ecc384_verify(IDevIdPubKey, IDevIdTbsDigest, IDevIdCertSig)`
+    `Result = ecc384_verify(IDevIdPubKeyEcdsa, IDevIdTbsDigestEcdsa, IDevIdCertSigEcdsa)`
+
+*(Note: Steps 7-10 are performed if Dual-Cert Mode is enabled)*
+
+7. Generate the `To Be Signed` DER Blob of the IDevId CSR with the MLDSA public key.
 
-7. Upload the CSR to mailbox and wait for JTAG to read the CSR out of the mailbox.
+    `IDevIdTbsMldsa = gen_tbs(IDEVID_CSR, IDevIdPubKeyMldsa)`
+
+8. Sign the IDevID `To Be Signed` DER Blob with IDevId MLDSA Private Key generated from the seed in Key Vault Slot 10.
+
+    `IDevIdTbsDigestMldsa = sha512_digest(IDevIdTbsMldsa)`
+
+    `IDevIdCertSigMldsa = mldsa87_sign(KvSlot10, IDevIdTbsDigestMldsa)`
+
+10. Verify the signature of IDevID `To Be Signed` Blob
+
+    `Result = mldsa87_verify(IDevIdPubKeyMldsa, IDevIdTbsDigestMldsa, IDevIdCertSigMldsa)`
+
+11. Upload the CSR(s) to mailbox and wait for JTAG to read the CSR out of the mailbox.
```

Comment:
nit: Don't need to solve in this PR, but we should add a structure for outputting multiple CSRs so the caller doesn't need to do ASN.1 parsing to separate them.

Reply:
Ack. I'll document the CSR format as part of the CSR envelop signing doc update PR.